lib/r10k/environment/with_modules.rb

```diff
@@ -90,31 +91,33 @@ def add_module(name, args)
     @managed_content[install_path] = Array.new unless @managed_content.has_key?(install_path)
 
     mod = R10K::Module.new(name, install_path, args, self.name)
-    mod.origin = 'Environment'
+    mod.origin = :environment
 
     @managed_content[install_path] << mod.name
     @modules << mod
   end
 
   def validate_no_module_conflicts
-    @puppetfile.load unless @puppetfile.loaded?
-    conflicts = (@modules + @puppetfile.modules)
+    puppetfile.load unless puppetfile.loaded?
+    conflicts = (@modules + puppetfile.modules)
                 .group_by { |mod| mod.name }
                 .select { |_, v| v.size > 1 }
-                .map(&:first)
     unless conflicts.empty?
-      conflicts_str = conflicts.join(', ')
+      conflicting_module_names = conflicts.keys.join(', ')
+      conflicting_puppetfile_modules = conflicts.values.flatten.reject { |mod| @modules.include?(mod) }
```

Comment:
I think it would be clearer if this `select`ed the conflicts in `puppetfile.modules` instead of `reject`ing those from `@modules`. I had to read this a few times to understand the intention.

Reply:
I'll flip it around